protocols/purple/purple.c

```diff
@@ -1133,7 +1158,12 @@ void prplcb_conv_new(PurpleConversation *conv)
 		/* libpurple brokenness: Whatever. Show that we join right away,
 		   there's no clear "This is you!" signaling in _add_users so
 		   don't even try. */
-		imcb_chat_add_buddy(gc, gc->ic->acc->user);
+		if (!set_getbool(&(ic->acc->set), "defer_joins")) {
+			imcb_chat_add_buddy(gc, gc->ic->acc->user);
+		}
+		else {
+			imcb_log(gc->ic, "Deferring join to %s", conv->name);
```

Comment:
This might be fine for developing/testing this feature but not very useful/meaningful in practice.